src/redux/fetchApi.js

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
+const uniqueId = 'EubIqMSX5vouKUWfEXQk';
+const URL = `${baseURL}/apps/${uniqueId}/books`;
+
+const initialState = {
+  books: [],
+  loading: false,
+};
+
+const GET_ALL_BOOKS = 'books/getAllBooks';
+const ADD_BOOK_API = 'book/addBookAPI';
+const REMOVE_BOOK_API = 'book/removeBookAPI';
+
+export const getAllBooks = createAsyncThunk(GET_ALL_BOOKS, async () => {
+  const respo = await axios.get(URL);
+  const { data } = respo;
+  const books = Object.entries(data).map(([key, value]) => {
+    const { title, category, author } = value[0];
+    return {
+      Id: key,
+      title,
+      author,
+      category,
+    };
+  });
+  return books;
+});
+
+export const addBookAPI = createAsyncThunk(ADD_BOOK_API, async (payload, thunkAPI) => {
+  await axios.post(URL, payload);
+  return thunkAPI.dispatch(getAllBooks());
+});
+
+export const removeBookAPI = createAsyncThunk(REMOVE_BOOK_API, async (payload, thunkAPI) => {
+  await axios.delete(`${URL}/${payload.id}`);
+  return thunkAPI.dispatch(getAllBooks());
+});
+
```

Comment:
- [ ] Your book delete implementation does not work. I am able to add books and the books are fetched from the API, but I cannot delete a book from the API. Kindly look into that and resolve it

Reply:
I fixed the functionality of removeBook. ThankYou for noticing